### ISSUE TITLE
[java] EmptyControlStatement should not allow empty try with concise resources

### ISSUE THREAD

**Issue by oowekyala** [pmd/pmd#4930]
**Affects PMD Version:** 7.0.0

**Rule:** [EmptyControlStatement](https://docs.pmd-code.org/latest/pmd_rules_java_codestyle.html#emptycontrolstatement)

**Description:**
Since #432 was closed EmptyControlStatement allows empty try-with-resources, provided the resources are named with a name like `ignored`.

I think this should only be done for resources that don't use shorthand form. For instance in our tests:
https://github.com/pmd/pmd/blob/f87f24ef9075ef0139a78d3a1ce90f9a976ee601/pmd-java/src/test/resources/net/sourceforge/pmd/lang/java/rule/codestyle/xml/EmptyControlStatement.xml#L138-L141

This looks like an anti pattern to me. If you're going to use an empty try statement for this you should write `in.close()` because it has the same effect but is clearer.

On the other hand the following should probably still be allowed:
https://github.com/pmd/pmd/blob/f87f24ef9075ef0139a78d3a1ce90f9a976ee601/pmd-java/src/test/resources/net/sourceforge/pmd/lang/java/rule/codestyle/xml/EmptyControlStatement.xml#L121-L122

Here an explicit `ignored` name is given to an expression. Even if it could be written
```java
ClientResponse response = execute(() -> target.request(mediaTypes).delete(), DELETE, new ExpectedResponse(status, required)));
response.close();
```
this code would pollute the enclosing scope with a useless variable name.

Another case to keep in mind is the following:
```java
try (in) {
} catch (Foo e) {
  // do something
}
```
If there is at least one catch block then it should be ok to keep the try block empty, even if the resource is unnamed.

Probably the rule should actually not care about whether the variables are named `ignored` or the like. If the variable is not used it will be flagged by UnusedVariable already.

**Expected outcome:**

PMD should report a violation at line ..., but doesn't. This is a false-negative.

**Running PMD through:** *[CLI | Ant | Maven | Gradle | Designer | Other]*
